### PR TITLE
Clarify that SQL transport V3 does not support Outbox with multi-inst…

### LINKED ISSUE #1817: SQL Server transport legacy multi-instance mode does not support the outbox
SQL Server transport upgrade guide (V2 -> V3) should mention that the legacy multi-instance mode in V3 does not support the Outbox, as opposed to V2 multi-instance.

The main doco page for the SQL Server transport contains this information but it should be easily accessible for people upgrading their endpoints.

### PR DESCRIPTION
**Pull request by MarcinHoppe**
Fixes #1817

### CODE REVIEW

File: nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 ### Multi-instance
 
-WARNING: The *multi-instance* option won't be supported in Versions 4 and higher. Version 3 supports *multi-instance* mode only with DTC.
+WARNING: The *multi-instance* option won't be supported in Versions 4 and higher.
```
Comment (indualagarsamy) [inline nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md]: >WARNING: The *multi-instance* option won't be supported in Versions 4 and higher.

should this be:

WARNING: The *multi-instance* option is no longer supported from Versions 4 and higher.

File: nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 
- * Requires DTC, or in case of Version 2 using Outbox and storing business data in the same database as Outbox data.
+WARNING: Version 3 supports *multi-instance* mode with DTC only. Outbox is not supported by Version 3 in *multi-instance* mode.
```
Comment (indualagarsamy) [inline nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md]: >WARNING: Version 3 supports *multi-instance* mode with DTC only. Outbox is not supported by Version 3 in *multi-instance* mode.

How about:

WARNING: Although Version 3 offers support for *multi-instance* mode when using DTC, Outbox is not a supported option when running in *multi-instance* mode.

Comment (SimonCropp) [inline nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md]: "And above"

File: nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/sqlserver/deployment-options_multi-instance_sqlserver_[,3].partial.md]: "And above"

**Pull request merged**